microcloud/service/service_handler.go

```diff
@@ -97,32 +98,44 @@ func (s *ServiceHandler) Start(state *state.State) error {
 		return err
 	}
 
+	networks, err := cloudMDNS.GetNetworkInfo()
+	if err != nil {
+		return err
+	}
+
 	info := cloudMDNS.ServerInfo{
 		Version:    cloudMDNS.Version,
 		Name:       s.Name,
-		Address:    s.Address,
 		Services:   services,
+		Networks:   networks,
 		AuthSecret: s.AuthSecret,
 	}
 
-	bytes, err := json.Marshal(info)
-	if err != nil {
-		return fmt.Errorf("Failed to marshal server info: %w", err)
-	}
+	for _, net := range info.Networks {
+		info.Address = net.Address
+		bytes, err := json.Marshal(info)
+		if err != nil {
+			return fmt.Errorf("Failed to marshal server info: %w", err)
+		}
 
-	s.server, err = cloudMDNS.NewBroadcast(s.Name, s.Address, s.Port, bytes)
-	if err != nil {
-		return err
+		server, err := cloudMDNS.NewBroadcast(s.Name, net.Address, s.Port, bytes)
```

Comment:
Just a bit confused as to why we need to have info.Networks in the struct we send over mDNS.

It makes sense that we need to send one broadcast per address on the system, but I don't understand why the payload of the broadcast has to include the addresses too. Can't the receive just tell what address send the broadcast?

Reply:
This way the mDNS lookup on just one network would report all of them. I suppose if there’s a lot of addresses it could be troublesome.

Reply:
One thing is that the mDNS lookup seems to get a bit slow with that many entries. 2 additional peers with 5 addresses each takes about 2 seconds to report all of them.

Reply:
I believe the mDNS records are slow because it keeps picking up previously-found records. If I'm right, then if we have a potential peer with significantly fewer addresses than the others, we might need to wait quite a while to get even a single record for it.

Maybe an API endpoint would be better suited for fetching the networks. So we just broadcast on every network and collect one record for each, then send an API request to each system to fetch all networks and filter down our peers based on that.

Reply:
Yeah, but if we only rely on the address we're receiving, then we're automatically filtering out all the addresses that we can't reach, that seems preferable to me.

Reply:
Otherwise we may end up showing and potentially attempting to use addresses which are not reachable or aren't directly on a shared subnet, which can then cause issues with OVN tunnels down the line.